### PR TITLE
feat: create Electron 5.0 blog post

### PR DESCRIPTION
**Pull request by BinaryMuse**
Getting this started early so that we can begin to properly document new features, breaking changes, and deprecations now instead of waiting till the last minute 😏 

Items to add:

- [x] Mixed sandboxed mode/pepper plugin/etc

### CODE REVIEW

File: data/blog/electron-5-0.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+Deprecated ServiceWorker APIs on WebContents in preparation for their removal. #16732
+Deprecated modules internally using remote.require in sandboxed renderer context. #15145
+Added webFrame.setIsolatedWorldInfo API and deprecated related APIs. #16932
+enableMixedSandbox and the --enable-mixed-sandbox command-line switch still exist for compatibility, but are deprecated and have no effect. #15894
+
+## App Feedback Program
+
+The [App Feedback Program](https://electronjs.org/blog/app-feedback-program) we instituted during the development of Electron 3.0 was successful, so we've continued it during the development of 5.0 as well. To learn more about the App Feedback Program and to participate in future betas, [check out our blog post about the program](https://electronjs.org/blog/app-feedback-program).
+
+## What's Next
```
Comment (sofianguy) [inline data/blog/electron-5-0.md]: I can get a 6.0.0 Timeline pushed up soon if we want to link to it in this section

File: data/blog/electron-5-0.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+Mixed-sandbox mode is now enabled by default. enableMixedSandbox and the --enable-mixed-sandbox command-line switch still exist for compatibility, but are deprecated and have no effect. #15894
+The default values of nodeIntegration and webviewTag are now false to improve security. #16235
+
+## Deprecations
+
+The following breaking changes are planned for Electron 6.0, and thus are deprecated in Electron 5.0.
+
+### Deprecation
+
```
Comment (sofianguy) [inline data/blog/electron-5-0.md]: Add note in this section or different section about EOL for 2-0-x line

Comment (sofianguy) [reply data/blog/electron-5-0.md]: reference here: https://electronjs.org/docs/tutorial/support#supported-versions

File: data/blog/electron-5-0.md (hunk 1 above)
Comment (sofianguy) [reply data/blog/electron-5-0.md]: https://github.com/electron/electron/pull/17866/files

File: data/blog/electron-5-0.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+## What's New?
+
+A large part of Electron's functionality is provided by Chromium, Node.js, and V8, the core components that make up Electron. As such, a key goal for the Electron team is to keep up with changes to these projects as much as possible, providing developers who build Electron apps access to new web and JavaScript features. To this end, Electron 5 features major version bumps to each of these components; Electron v5.0.0 includes Chromium `73.0.3683.117`, Node `12.0.0`, and V8 `7.3.492.27`.
+
+In addition, Electron 5 includes changes to Electron-specific APIs. You can find a summary of the major changes in Electron 5 below; for the full list of changes, check out the [Electron v5.0.0 release notes](https://github.com/electron/electron/releases/tag/v5.0.0).
+
+### Promisification
+
+Electron 5 contains changes as part of our [Promisification](https://github.com/electron/electron/blob/master/docs/api/modernization/promisification.md) initiative to convert callback-based functions in Electron to return Promises.  The following APIs were converted for Electron
```
Comment (MarshallOfSound) [inline data/blog/electron-5-0.md]: This link should point to a commit hash or the Electron 5 branch

File: data/blog/electron-5-0.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+5:
```
Comment (MarshallOfSound) [inline data/blog/electron-5-0.md]: Why is `5:` on it's own line here?

File: data/blog/electron-5-0.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+Mixed sandbox mode is now enabled by default.  Renderers launched with `sandbox: true`
+will now be actually sandboxed, where previously they would only be sandboxed if mixed-sandbox mode was also enabled.
+
+### Security improvements
+The default values of `nodeIntegration` and `webviewTag` are now false to improve security.
+
+### Spellchecker now asynchronous
+
+The SpellCheck API has been changed to provide asynchronous results.
```
Comment (MarshallOfSound) [inline data/blog/electron-5-0.md]: We should link to an implementation or the API here to help people migrate.  cc @nitsakh

File: data/blog/electron-5-0.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+## Deprecations
+
+The following breaking changes are planned for Electron 6.0, and thus are deprecated in Electron 5.0.
+
+### Mksnapshot binaries for arm and arm64
+Native binaries of mksnapshot for arm and arm64 are deprecated and will be removed in 6.0.0.  Snapshots can be created for arm and arm64 using the x64 binaries.
+
+### ServiceWorker APIs on WebContents
+Deprecated ServiceWorker APIs on WebContents in preparation for their removal.
```
Comment (MarshallOfSound) [inline data/blog/electron-5-0.md]: Can we list them here, people shouldn't have to google "electron service worker APIs", this currently just raises more questions

File: data/blog/electron-5-0.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+### Automatic modules with sandbox
```
Comment (MarshallOfSound) [inline data/blog/electron-5-0.md]:
```suggestion
### Automatic modules with sandboxed webContents
```

File: data/blog/electron-5-0.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+In order to improve security, the following modules are being deprecated for use directly via `require` and will instead need to be included via `remote.require`:
```
Comment (MarshallOfSound) [inline data/blog/electron-5-0.md]:
```suggestion
In order to improve security, the following modules are being deprecated for use directly via `require` and will instead need to be included via `remote.require` in a sandboxed webcontents:
```

File: data/blog/electron-5-0.md (hunk 7 above)
Comment (ckerr) [reply data/blog/electron-5-0.md]: Fixed 0d3e42512eb4ee27ad0ed21a37df944f7460d2f4

File: data/blog/electron-5-0.md (hunk 4 above)
Comment (ckerr) [reply data/blog/electron-5-0.md]: FIxed 05c822a04ce3d87ed91d45a230e0d295d97c02f1

File: data/blog/electron-5-0.md (hunk 5 above)
Comment (ckerr) [reply data/blog/electron-5-0.md]: FIxed 385d4ef62a17068b269308eb0fabf41af8a0031f

File: data/blog/electron-5-0.md (hunk 6 above)
Comment (ckerr) [reply data/blog/electron-5-0.md]: Fixed 12f833c310d21e7525ebb3556fe833ed343594de

File: data/blog/electron-5-0.md (hunk 3 above)
Comment (ckerr) [reply data/blog/electron-5-0.md]: Fixed dab62bf22db34855aacd3f9b6805dd093f11eedc

File: data/blog/electron-5-0.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+Native binaries of mksnapshot for arm and arm64 are deprecated and will be removed in 6.0.0. Snapshots can be created for arm and arm64 using the x64 binaries.
+
+### ServiceWorker APIs on WebContents
+Deprecated ServiceWorker APIs on WebContents in preparation for their removal.
+* `webContents.hasServiceWorker`
+* `webContents.unregisterServiceWorker`
+
+### Automatic modules with sandboxed webContents
+In order to improve security, the following modules are being deprecated for use directly via `require` and will instead need to be included via `remote.require` in a sandboxed webcontents:
+* `electron.screen`
```
Comment (miniak) [inline data/blog/electron-5-0.md]: this one is available via `electron.remote.screen`

File: data/blog/electron-5-0.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+### Mksnapshot binaries for arm and arm64
+Native binaries of mksnapshot for arm and arm64 are deprecated and will be removed in 6.0.0. Snapshots can be created for arm and arm64 using the x64 binaries.
+
+### ServiceWorker APIs on WebContents
+Deprecated ServiceWorker APIs on WebContents in preparation for their removal.
+* `webContents.hasServiceWorker`
+* `webContents.unregisterServiceWorker`
+
+### Automatic modules with sandboxed webContents
+In order to improve security, the following modules are being deprecated for use directly via `require` and will instead need to be included via `remote.require` in a sandboxed webcontents:
```
Comment (miniak) [inline data/blog/electron-5-0.md]: this change is more about making the use of the `remote` module explicit due to security and performance implications.

File: data/blog/electron-5-0.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+* `shell.openExternal`
+* `webContents.loadFile`
+* `webContents.loadURL`
+* `webContents.zoomLevel`
+* `webContents.zoomFactor`
+* `win.capturePage`
+
+### System colors access for macOS
+
+These functions were changed or added to `systemPreferences` to access macOS systems colors:
```
Comment (malept) [inline data/blog/electron-5-0.md]:
```suggestion
These functions were changed or added to `systemPreferences` to access macOS systems' colors:
```

Alternatively:

```suggestion
These functions were changed or added to `systemPreferences` to access macOS's system colors:
```

File: data/blog/electron-5-0.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+### Defaults for packaged apps
+
+Packaged apps will now behave the same as the default app: a default application menu will be created unless the app has one and the `window-all-closed` event will be automatically handled unless the app handles the event.
+
+### Mixed sandbox
+
+Mixed sandbox mode is now enabled by default. Renderers launched with `sandbox: true` will now be actually sandboxed, where previously they would only be sandboxed if mixed-sandbox mode was also enabled.
+
+### Security improvements
+The default values of `nodeIntegration` and `webviewTag` are now false to improve security.
```
Comment (malept) [inline data/blog/electron-5-0.md]:
```suggestion
The default values of `nodeIntegration` and `webviewTag` are now `false` to improve security.
```

File: data/blog/electron-5-0.md (hunk 9 above)
Comment (ckerr) [reply data/blog/electron-5-0.md]: What change would you suggest ?

Comment (ckerr) [reply data/blog/electron-5-0.md]: @miniak what change would you suggest here?

**Pull request merged**